### ISSUE TITLE
Other Dotnet Support

### ISSUE THREAD

**Issue by cschuchardt88** [neo-project/neo#2898]
**Summary or problem description**
Neo needs to be split apart into different libraries. Also build in dotnet standard or make SDKs for Unity and .Net Maui and other dotnet versions. Dotnet standard would be ideal here. Kestrel is the big part of the problem here. Doing this will allow us to pick and choose what we need and have the flexibility pick the dotnet flavors we want to use. This can open a whole new world of possibilities for The Neo Project to expand. Well Unity should be a no brainer, "Games". In addition there are lots of people that want to make cross platform phone/tablet/computer app as well.

**Do you have any solution you want to propose?**
Split up logic for non-dotnet standard.  Allow basic logic to make transactions and sign them and interact with smart contracts.

**Split up**
- UInt160 / UInt256 / Etc (_Neo P2P Payloads_)
- Neo.vm.dll (_I think it supports dotnet standard_)
- Wallet API
- Cryptography
- Plugins
- Smart Contract
- Persistence
- Network/P2P
- Ledger
- Neo.Json (_should be removed and use something better_)
- NeoSystem

**Neo Version**
- Neo 3.7.0

**Where in the software does this update applies to?**
- CLI
- Plugins
- Ledger
- P2P (TCP)
- SDK
- VM

**Comment by vncoelho** [comment]
@cschuchardt88, I understand the line of reasoning, but I think that this would be a major version like 4.0.0 because it would involve several critical components that would require extensive tests.

**Comment by cschuchardt88** [comment]
Yes you right. but i think actually doing it will be the big hurdle. I think moving Kestrel to separate library will fix a lot of the problems. But maybe adding logic to code like below will be best for longevity of the project. Would should at least get in the habit of writing code like that. Instead of having couple of people rebuild everything.
```csharp
#if DOTNET_STANDARD
// do something
#endif
```
